File: tools/bin2array.py
#!/usr/bin/env python

import sys

if len(sys.argv) != 2:
  print "Usage: bin2array.py <filename>"
  sys.exit(0)

print "  static byte[] array ="
print "  {"

count = 0
s = ""

fp = open(sys.argv[1], "rb")

while 1:
  data = fp.read(4096)
  if len(data) == 0: break

  for c in data:
    byte = ord(c)

    if byte > 127: byte = -((byte ^ 0xff) + 1)

    s += " %4d," % byte
    count += 1

    if (count % 8) == 0:
      print "    " + s
      count = 0
      s = ""

  if s != "": print "    " + s

fp.close()

print "  };"

File: tools/bin2array.rs
use std::env;
use std::fs::File;
use std::io::prelude::*;

fn main()
{
  let args: Vec<String> = env::args().collect();
  let filename = &args[1];

  let mut file = File::open(filename).expect("File not found");

  let mut buf = [0u8; 65536];
  let bytes_read = file.read(&mut buf).unwrap();

  println!("  static byte[] array =");
  println!("  {{");

  let mut i = 0;

  while i < bytes_read
  {
    let mut out = String::from("    ");
    let mut count = bytes_read - i;

    if count > 8 { count = 8; }

    for n in 0..count
    {
      let mut next = format!(" {:4.0},", buf[i + n] as i8);
      out.push_str(&mut next);
    }

    println!("{}", out);

    i += 8
  }

  println!("  }};")
}

